### PR TITLE
test(NODE-4917): remove grouped callback style tests

### PR DESCRIPTION
**Pull request by nbbeeken**
#### What is changing?

Removing a large amount of callback style tests.

#### What is the motivation for this change?

Prep for callback support removal

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: test/integration/node-specific/operation_examples.test.ts [hunk 1]
```diff
@@ -1785,13 +1779,13 @@ describe('Operation (Promises)', function () {
             o
           )
           .then(function (result) {
-            test.ok(result);
+            expect(result).to.exist;
```
Comment (durran) [inline test/integration/node-specific/operation_examples.test.ts:1782]: I can't seem to comment on the exact line but line 1777 is failing lint due to the empty function.

File: test/integration/node-specific/operation_examples.test.ts [hunk 2]
```diff
@@ -1836,7 +1830,7 @@ describe('Operation (Promises)', function () {
             return collection.createIndex({ title: 1 }, { unique: true });
           })
           .then(function (indexName) {
-            test.ok(indexName);
+            expect(indexName).to.exist;
```
Comment (durran) [inline test/integration/node-specific/operation_examples.test.ts:1833]: Same with lint for empty function on line 1827.

File: test/integration/crud/server_errors.test.js [hunk 3]
```diff
@@ -86,30 +87,20 @@ describe('Errors', function () {
...
-      c.insertOne({ a: 2, b: 5 }, { writeConcern: { w: 1 } }, err => {
-        expect(err).to.not.exist;
-        c.findOne({ a: 2 }, { projection: { a: 1, b: 0 } }, err => {
-          expect(PROJECTION_ERRORS).to.include(err.errmsg);
-          done();
-        });
-      });
-    }
+  it('should return an error object with message when mixing included and excluded fields', async () => {
+    const db = client.db(this.configuration.db);
```
Comment (durran) [inline test/integration/crud/server_errors.test.js:91]: When using arrow functions in mocha we lose the proper `this` reference so `this.configuration` is undefined.

File: test/integration/crud/server_errors.test.js [hunk 4]
```diff
@@ -86,30 +87,20 @@ describe('Errors', function () {
...
-    test: function (done) {
-      const db = client.db(this.configuration.db);
-      const c = db.collection('test_error_object_should_include_message');
-      c.findOne({}, { projection: { a: 1, b: 0 } }, err => {
-        expect(PROJECTION_ERRORS).to.include(err.errmsg);
-        done();
-      });
-    }
+  it('should reject promise with projection errors', async () => {
+    const db = client.db(this.configuration.db);
```
Comment (durran) [inline test/integration/crud/server_errors.test.js:100]: When using arrow functions in mocha we lose the proper `this` reference so `this.configuration` is undefined.

**Review by durran — CHANGES_REQUESTED** [review]

File: test/integration/crud/server_errors.test.js (hunk 4 above)
Comment (nbbeeken) [reply test/integration/crud/server_errors.test.js]: done

File: test/integration/crud/server_errors.test.js (hunk 3 above)
Comment (nbbeeken) [reply test/integration/crud/server_errors.test.js]: done

File: test/integration/node-specific/operation_examples.test.ts (hunk 2 above)
Comment (nbbeeken) [reply test/integration/node-specific/operation_examples.test.ts:1833]: done

File: test/integration/node-specific/operation_examples.test.ts (hunk 1 above)
Comment (nbbeeken) [reply test/integration/node-specific/operation_examples.test.ts:1782]: done

**Review by durran — CHANGES_REQUESTED** [review]
Looks like this now has introduced a connection leak somewhere in the tests perhaps?

**Comment by nbbeeken** [comment]
Missed one of the old `test.` assertions, 🤞 hopefully that was all

File: test/integration/node-specific/operation_examples.test.ts [hunk 5]
```diff
@@ -3394,103 +3397,94 @@ describe('Operation (Promises)', function () {
...
-          })
-          .catch(function (err) {
-            test.ok(err instanceof Error);
-            test.equal(
-              `Profiling level must be one of "${enumToString(ProfilingLevel)}"`,
-              err.message
-            );
-            return client.close();
-          });
+  it('setProfilingLevel changes profiling level', async function () {
```
Comment (durran) [inline test/integration/node-specific/operation_examples.test.ts:3403]: Looks like this one is still failing as it's should only be running on a single and is failing CI on other topologies.

**Review by durran — CHANGES_REQUESTED** [review]

**Comment by nbbeeken** [comment]
Ok finally in the clear 😅  (remaining errors are all known external ones)

**Review by durran — DISMISSED** [review]

File: test/integration/collection-management/promise_collection_db_management.test.ts [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import { assert as test, setupDatabase } from '../shared';
```
Comment (baileympearson) [inline test/integration/collection-management/promise_collection_db_management.test.ts:1]: let's rename this file to get rid of `promise` 😄

File: test/integration/crud/bulk.test.ts [hunk 7]
```diff
@@ -107,37 +107,35 @@ describe('Bulk', function () {
...
-      const bulk = db
-        .collection('unordered_bulk_promise_form')
-        .initializeUnorderedBulkOp({ writeConcern: { w: 1 } });
-      bulk.insert({ a: 1 });
-
-      const r = await bulk.execute();
-      test.ok(r);
-      test.deepEqual({ w: 1 }, bulk.s.writeConcern);
-    });
+  it('Should correctly execute unordered bulk operation in promise form', async function () {
```
Comment (baileympearson) [inline test/integration/crud/bulk.test.ts:110]:
```suggestion
  it('Should correctly execute unordered bulk operation', async function () {
```

File: test/integration/crud/bulk.test.ts [hunk 8]
```diff
@@ -107,37 +107,35 @@ describe('Bulk', function () {
...
-      const bulk = db
-        .collection('unordered_bulk_promise_form')
-        .initializeOrderedBulkOp({ writeConcern: { w: 1 } });
-      bulk.insert({ a: 1 });
-
-      const r = await bulk.execute();
-      test.ok(r);
-      test.deepEqual({ w: 1 }, bulk.s.writeConcern);
-    });
+  it('Should correctly execute ordered bulk operation in promise form', async function () {
```
Comment (baileympearson) [inline test/integration/crud/bulk.test.ts:122]:
```suggestion
  it('Should correctly execute ordered bulk operation, async function () {
```

File: test/integration/crud/find_and_modify.test.js [hunk 9]
```diff
@@ -9,85 +9,83 @@ describe('Find and Modify', function () {
...
-              test.equal(1, r.value.b);
-
-              client.close(done);
-            })
-            .catch(function (err) {
-              test.ok(err != null);
-            });
-        });
-      });
+  it('Should correctly execute findOneAndDelete operation With Promises and no options passed in', function (done) {
```
Comment (baileympearson) [inline test/integration/crud/find_and_modify.test.js:12]:
```suggestion
  it('Should correctly execute findOneAndDelete operation and no options passed in', function (done) {
```

File: test/integration/crud/find_and_modify.test.js [hunk 10]
```diff
@@ -9,85 +9,83 @@ describe('Find and Modify', function () {
...
-              test.equal(1, r.value.b);
-
-              client.close(done);
-            })
-            .catch(function (err) {
-              test.ok(err != null);
-            });
-        });
-      });
+  it('Should correctly execute findOneAndUpate operation With Promises and no options passed in', function (done) {
```
Comment (baileympearson) [inline test/integration/crud/find_and_modify.test.js:39]:
```suggestion
  it('Should correctly execute findOneAndUpate operation and no options passed in', function (done) {
```

File: test/integration/crud/find_and_modify.test.js [hunk 11]
```diff
@@ -9,85 +9,83 @@ describe('Find and Modify', function () {
...
-        const col = db.collection('find_one_and_replace_with_promise_no_option');
-        col.insertMany([{ a: 1, b: 1 }], { writeConcern: { w: 1 } }).then(function (r) {
-          expect(r).property('insertedCount').to.equal(1);
-
-          col
-            .findOneAndReplace({ a: 1 }, { a: 1 })
-            .then(function (r) {
-              test.equal(1, r.lastErrorObject.n);
-              test.equal(1, r.value.b);
+  it('Should correctly execute findOneAndReplace operation With Promises and no options passed in', function (done) {
```
Comment (baileympearson) [inline test/integration/crud/find_and_modify.test.js:66]:
```suggestion
  it('Should correctly execute findOneAndReplace operation and no options passed in', function (done) {
```

File: test/integration/index_management.test.js [hunk 12]
```diff
@@ -8,51 +8,49 @@ describe('Indexes', function () {
...
-        // Create an index
-        client
-          .db(configuration.db)
-          .createIndex('promiseCollectionCollections1', { a: 1 })
-          .then(function (r) {
-            test.ok(r != null);
-
-            client.close(done);
-          });
+  it('Should correctly execute createIndex using Promise', {
```
Comment (baileympearson) [inline test/integration/index_management.test.js:11]:
```suggestion
  it('Should correctly execute createIndex', {
```

**Review by baileympearson — CHANGES_REQUESTED** [review]

File: test/integration/collection-management/promise_collection_db_management.test.ts (hunk 6 above)
Comment (nbbeeken) [reply test/integration/collection-management/promise_collection_db_management.test.ts:1]: done

**Review by baileympearson — APPROVED** [review]

**Pull request merged**